### PR TITLE
Wagtail 5.1 upgrades and test workflow update

### PR DESCRIPTION
**Pull request by katdom13**
Upstream PR: https://github.com/labd/wagtail-2fa/pull/227

### CODE REVIEW

File: sandbox/sandbox/settings.py
Comment (nickmoreton) [inline sandbox/sandbox/settings.py:49]: I'm not sure what this means, sorry.

File: tests/conftest.py
Comment (nickmoreton) [inline tests/conftest.py:33]: Ditto

**Review by nickmoreton — APPROVED** [review]
I approve this but left a question. Thanks

File: tests/conftest.py
Comment (katdom13) [reply tests/conftest.py:33]: Hi @nickmoreton ,
I based it on Jazzband's update here:
https://github.com/jazzband/wagtailmenus/pull/461/files#diff-e285c1b9901cd202e89d9f37e3249a6fee524a65428895e1a45fd062cc0f9acaR22-R23

Which is a resolution for this Wagtail 5.1 upgrade consideration: [wagtail.contrib.modeladmin is deprecated](https://docs.wagtail.org/en/stable/releases/5.1.html#wagtail-contrib-modeladmin-is-deprecated)

I updated the PR to actually replace the installed app depending on the Wagtail version.
https://github.com/torchbox-forks/wagtail-2fa/pull/3/commits/b00c2b74b69947cbfc90faa2782dccaa65a8d65c

**Comment by ArnarTumi** [comment]
Hey, any chance that this PR and PR#2 could be merged? If not, any chance that you could add the line from #2 to this PR @katdom13 so that we could use this branch as temporary solution for WT5.1 until it will be merged?

**Review by Stormheg — CHANGES_REQUESTED** [review]
Please note that this is not actually compatible with Wagtail 5.1!

The [`otp_form.html`](https://github.com/torchbox-forks/wagtail-2fa/blob/ec7650ecdf7d040ae9bc23ac4411ba2cb929a5fd/src/wagtail_2fa/templates/wagtail_2fa/otp_form.html#L58) template includes an SVG logo that was deleted in https://github.com/wagtail/wagtail/commit/ecb2126ceaa8e4697963babfccb82fad613c63a4

I've done a quick 'n dirty fix on my fork here: https://github.com/techonomydev/wagtail-2fa/commit/7fe52ba43788fd00352ce19c79ea777b06d72f69 but you may want to create proper fix because my fix breaks support for Wagtail 5.0 and older.

**Comment by katdom13** [comment]
> Please note that this is not actually compatible with Wagtail 5.1!
> 
> The [`otp_form.html`](https://github.com/torchbox-forks/wagtail-2fa/blob/ec7650ecdf7d040ae9bc23ac4411ba2cb929a5fd/src/wagtail_2fa/templates/wagtail_2fa/otp_form.html#L58) template includes an SVG logo that was deleted in [wagtail/wagtail@ecb2126](https://github.com/wagtail/wagtail/commit/ecb2126ceaa8e4697963babfccb82fad613c63a4)
> 
> I've done a quick 'n dirty fix on my fork here: [techonomydev@7fe52ba](https://github.com/techonomydev/wagtail-2fa/commit/7fe52ba43788fd00352ce19c79ea777b06d72f69) but you may want to create proper fix because my fix breaks support for Wagtail 5.0 and older.

Hi @Stormheg ,
Thanks for checking! How about this workaround? https://github.com/torchbox-forks/wagtail-2fa/pull/3/commits/695027bcffc2dbda5b57b67f1224f775d107b21f

@ArnarTumi ,
I have merged the other PR. 

Thanks!

**Comment by Stormheg** [comment]
Your fix looks good @katdom13 👍

What would be even better is if you could add a test that checks if the otp form page renders okay without crashing. A smoke test.

That way we will be alerted if Wagtail changes one if its internal template includes again. What do you think?

**Comment by nickmoreton** [comment]
> Hey, any chance that this PR and PR#2 could be merged? If not, any chance that you could add the line from #2 to this PR @katdom13 so that we could use this branch as temporary solution for WT5.1 until it will be merged?

@ArnarTumi This has been rebased now on the maintainers master branch so `"wagtailadmin_sprite",` has been added in.

**Comment by nickmoreton** [comment]
@katdom13 All good thanks and tested locally using the sandbox

**Comment by katdom13** [comment]
@Stormheg , @nickmoreton ,
Will be adding test in another PR. Merging this for now.

**Pull request merged**
